tests/semantic_analyser.cpp

```diff
@@ -3973,6 +3982,130 @@ TEST_F(SemanticAnalyserTest, tuple_nested)
+                               Tuple({ Cast(Typeof(SizedType(Type::integer)),
+                                            TupleAccess(Variable("$b"), 0)),
+                                       TupleAccess(Variable("$b"), 1) })),
+            Jump(ast::JumpType::RETURN) })) });
+  test(
+      R"(begin { $a = ((int16)1, "hi"); $b = ((uint16)2, "hellostr"); $b = $a })",
+      ExpectedAST{ Program().WithProbe(Probe(
+          { "begin" },
+          { AssignVarStatement(Variable("$a"),
+                               Tuple({ Cast(Typeof(SizedType(Type::integer)),
+                                            Integer(1)),
+                                       String("hi") })),
+            AssignVarStatement(
+                Variable("$b"),
+                Tuple(
+                    { Cast(Typeof(SizedType(Type::integer)),
+                           Cast(Typeof(SizedType(Type::integer)), Integer(2))),
+                      String("hellostr") })),
+            AssignVarStatement(Variable("$b"),
+                               Tuple({ Cast(Typeof(SizedType(Type::integer)),
+                                            TupleAccess(Variable("$a"), 0)),
+                                       Cast(Typeof(SizedType(Type::string)),
+                                            TupleAccess(Variable("$a"), 1)) })),
+            Jump(ast::JumpType::RETURN) })) });
+  test(
+      R"(begin { @a = ((int16)1, "hi"); @b = ((uint16)2, "hellostr"); @a = @b })",
+      ExpectedAST{ Program().WithProbe(Probe(
+          { "begin" },
+          { AssignMapStatement(
+                Map("@a"),
+                Integer(0),
+                Tuple(
+                    { Cast(Typeof(SizedType(Type::integer)),
+                           Cast(Typeof(SizedType(Type::integer)), Integer(1))),
+                      Cast(Typeof(SizedType(Type::string)), String("hi")) })),
+            AssignMapStatement(Map("@b"),
+                               Integer(0),
+                               Tuple({ Cast(Typeof(SizedType(Type::integer)),
+                                            Integer(2)),
+                                       String("hellostr") })),
+            AssignMapStatement(
+                Map("@a"),
+                Integer(0),
+                Tuple({ Cast(Typeof(SizedType(Type::integer)),
+                             TupleAccess(MapAccess(Map("@b"), Integer(0)), 0)),
+                        TupleAccess(MapAccess(Map("@b"), Integer(0)), 1) })),
+            Jump(ast::JumpType::RETURN) })) });
+  test(
+      R"(begin { print(if (pid == 1) { ((int16)1, "hi") } else { ((uint16)2, "hellostr") }); })",
+      ExpectedAST{ Program().WithProbe(Probe(
+          { "begin" },
+          { ExprStatement(Block(
+              { ExprStatement(Call(
+                    "print",
+                    { If(Binop(Operator::EQ, Builtin("pid"), Integer(1)),
+                         Tuple({ Cast(Typeof(SizedType(Type::integer)),
```

Comment:
I think that you want something here that matches against the size? You just need to watch that matcher hook. (Here and elsewhere, .. right now you are just checking that a integer cast is injected, not the specific size.)

Reply:
Ah ok, I missed that. Will add.